[1주차 기본과제] 장정안의 제주도일기 ㅋㅋ

ExceptAnyone opened the pull request:
<!-- PR의 제목은 "[ n주차 기본/심화/생각 과제 ] 과제명 " 으로 작성해주시면 되겠습니다 -->
<!-- PR은 리뷰어를 위한 글입니다, 보다 더 상세하게 적음으로써 소통을 활성화해봐요! -->

## ✨ 구현 기능 명세


- [x] header

주제를 담은 제목을 넣어줍니다.

- [x] section (preview)

다양한 사진들을 미리볼 수 있는 섹션입니다.

flex를 사용하여 일정한 간격으로 배치합니다.
width값을 벗어나면 가로로 스크롤이 되도록 구현합니다.
 nav

- [x] 목차

각 목차 hover시 underline과 색상 변경이 일어납니다.
목차를 클릭하면 해당 목차가 있는 곳으로 이동합니다.
 section (img)

각 목차에 관련된 사진들이 있습니다

목차는 sticky로 화면 상단에 닿으면 고정되도록 구현합니다.

- [x] 첫번째 섹션
flex를 이용해서 같은 크기의 사진들로 나열합니다.
width를 넘으면 다음줄로 넘어가도록 구현합니다 (flex-wrap)

- [x] 두번째 섹션
flex 를 사용하여 위와 같은 형태로 배치합니다.

- [ ]세번째 섹션
grid를 사용하여 위와 같은 형태로 배치합니다.

 - [x] footer

마무리 말 또는 다양한 정보를 자유롭게 넣으셔도 됩니다!

 우측 하단에 top 버튼을 넣습니다.

top 버튼을 누르면 최상단으로 이동합니다.


<br />

## 💎 PR Point

- 항상 px 단위만 사용하다가 이번에, rem이라는 단위를 처음 써봤습니다.
```css
.group-img {
  width: 40rem;
  height: 40rem;
  margin: 5px;`
```

- preview에서는 overflow를 이용하여 가로스크롤을 구현하였습니다.
```css
.group {
  display: flex;
  overflow: auto;
  padding-top: 10px;
}
```

<br />

- 목차
마찬가지로 hover에서도 transition 속성을 처음 사용해보았습니다.
```css
.nav_list:hover {
  color: skyblue;
  transition: all 250ms ease-out;
  left: 0%;
  width: 100%;
  text-decoration: underline;
};
```

<br />

- 목차 클릭 시 이동하도록 구현하였습니다.
```javascript
<ul>
          <li class="nav_list"><a href="#food">음식</a></li>
          <li class="nav_list"><a href="#house">숙소</a></li>
          <li class="nav_list"><a href="#etc">기타</a></li>
        </ul> 
```

<br />


- [ ] 첫 번째 섹션

 - 알아서 다음 줄로 바꿔주기 위해  flex-wrap: wrap;을 사용하였습니다


```css
.food_image {
  width: 30rem;
  height: 30rem;
  margin: 5px;
  flex-wrap: wrap;
}
```
<br />

- [ ] 두 번째 섹션
 -  1. 왼쪽 컨테이너에 이미지 1개, 오른쪽 컨테이너에 나머지 이미지를 배치하였습니다
 ```html
<div class="left_container">
            <img
              src="./img/img11.JPG"
              alt="숙소 창문"
              class="house_image_left"
            />
            </div>
            <div class="right_container">
              <img src="./img/img12.JPG" alt="숙소 외경" class="house_image" />
              <img src="./img/img20.JPG" alt="숙소 정문" class="house_image" />
              <img src="./img/img21.JPG" alt="숙소 앞 밭" class="house_image" />
              <img src="./img/img19.JPG" alt="카페 앞 바다3" class="house_image" />
            </div>
          </div>
```

- 전체로 묶은 div태그에 flex를 주고, 오른쪽 컨테이너에 flex-wrap을 주었습니다.
``` css
.house_image_all {
  display: flex;
  height: 100%;
}

.house_image_left {
  width: 100%;
  height: 100%;
}

.right_container {
  flex-wrap: wrap;
  height: 100%;
}
```

<br />

- [ ] 세 번째 섹션
 1. 4행 6열로 설정하였습니다. 
 ```css
.grid_container {
  display: grid;
  grid-template-rows: repeat(4, 1fr);
  grid-template-columns: repeat(6, 1fr);
  gap: 10px;
  height: 100vh;
}
```

2. 그 다음 이미지의 위치를 생각하여 배치하였습니다.
```css
.etc_image:nth-child(1) {
  grid-row: 1 / 2;
  grid-column: 1 / 4;
  width: 100%;
  height: calc(100vh / 4 - 10px);
}

.etc_image:nth-child(2) {
  grid-row: 2 / 3;
  grid-column: 1 / 4;
  width: 100%;
  height: calc(100vh / 4 - 10px);
}

.etc_image:nth-child(3) {
  grid-row: 1 / 3;
  grid-column: 4 / 7;
  width: 100%;
  height: calc(100vh / 2 - 10px);
}

.etc_image:nth-child(4) {
  grid-row: 4 / 5;
  grid-column: 1 / 5;
  width: 100%;
  height: calc(100vh / 2 - 10px);
}

.etc_image:nth-child(5) {
  grid-row: 4 / 5;
  grid-column: 5 / 7;
  width: 100%;
  height: calc(100vh / 2 - 10px);
}
```

        


## 🥺 소요 시간, 어려웠던 점

- `12h`
- 살면서 그리드를 처음 써봤습니다. 아름이가 알려준 사이트를 정독한 후, 나중에 들어야겠다고 구매해놓은 1분코딩의 flex와 grid강의를 이번 세미나를 통해 조금이나마 참고하게 되었습니다. 

<br />

## 🌈 구현 결과물
https://github.com/DO-SOPT-WEB/JeonganJang/assets/126966681/d160ff2a-af93-4cc0-b0ee-789768db1501https://github.com/DO-SOPT-WEB/JeonganJang/assets/126966681/d160ff2a-af93-4cc0-b0ee-789768db1501

<!-- 스크린샷, gif, 배포링크 등 자유롭게 작성하되 애니메이션이나 동적UI 확인이 필요한 경우에는 스크린샷은 인정하지 않습니다. -->

ExceptAnyone committed:
init: 초기세팅

ExceptAnyone committed:
feat: 헤더 생성

ExceptAnyone committed:
chore: img/video 파일 추가

ExceptAnyone committed:
chore: 이미지 파일 추가 업로드

ExceptAnyone committed:
feat: 헤더 설정 및 가운데 이미지 가로스크롤, nav바 설정

헤더를 검정색 바탕에 흰색 글씨로 설정하고, 화면 중간 섹션에 가로스크롤을 구현하였으며,
nav_list를 두고 hover시 색상변경과 밑줄이 그어지게 하였습니다

ExceptAnyone committed:
chore: 필요없는 이미지 삭제

ExceptAnyone committed:
style: nav_title 중간 이동

ExceptAnyone committed:
nav_list에 a태그 추가

ExceptAnyone committed:
style: 링크에 밑줄과 색상 제거

ExceptAnyone committed:
feat: 목차 클릭시 해당 섹션으로 이동

ExceptAnyone committed:
style: 첫 컨텐츠 섹션 헤더 부분 상단 고정 및 해당 이미지 사이즈 조절

ExceptAnyone committed:
chore: img 파일 추가

음식 사진이 부족하여 관련 이미지 파일을 추가하였습니다.

ExceptAnyone committed:
feat: 두 번째, 세 번째 섹션 기본틀 구현
(추후 flex,grid 이용하여 배치 예정)

추후에 flex,grid 이용하여 배치하겠습니다. 우선적으로 사진만 배치 해놓았습니다.

ExceptAnyone committed:
feat: 두 번째 섹션 배치 구현

ExceptAnyone committed:
feat: footer 설정 및 상단 이동 버튼 생성

세 번째 섹션인 grid 관련해서 일부 주석처리 후 다시 작업 예정입니다.

ExceptAnyone committed:
style: 세 번째 섹션 grid형식 배치

ExceptAnyone committed:
comment: 내가 css를 못하는게 버그다

Arooming commented on week1/index.html:
```diff
@@ -0,0 +1,76 @@
+        <img src="./img/img9.JPG" alt="단체사진2" class="group-img" />
+        <img src="./img/img3.JPG" alt="단체사진3" class="group-img" />
+        <img src="./img/img17.JPG" alt="단체사진4" class="group-img" />
+        <img src="./img/img23.jPG" alt="단체사진5" class="group-img" />
```
이번 과제에서 검색엔진 최적화에 대해서 읽으면서 알게 된 사실인데..!!
웹 접근성을 높이기 위해서도, 검색엔진 최적화를 위해서도 alt는 구체적으로 적어주는게 좋다고 하더라구! (대신 너무 길지 않게 ..!)
다음 과제부턴 구체적으로 적어주는거 어떨까아!! 

+) 그리고 찾아보니까 스크린 리더에서 `<img>` 자체를 사진으로 바꿔서 읽어주기 때문에 alt 태그 안에 `사진`이라는 말이 들어가게 되면 `단체 사진1 사진` 이렇게 읽어준다고 하네?! 크크크 사진이라는 말도 빼주면 좋을듯 ~~

ExceptAnyone replied:
오오 바로 수정했슴다 :)

Arooming commented on week1/index.html:
```diff
@@ -0,0 +1,76 @@
+      </nav>
+      <section id="food" class="first_content">
+        <h2 class="first_content_header">food~</h2>
+        <div>
```
요기 section에 들어간 가장 바깥쪽 div들은 article 태그로 바꿔줘도 좋을 것 같다는 생각..!!
article 태그가 하나의 주제를 묶는 독립적인 컨텐츠를 나타내니까 의미가 맞는 것 같넹 !!

Arooming commented on week1/index.html:
```diff
@@ -0,0 +1,76 @@
+      </section>
+      <section id="etc">
+        <h2 class="third_content_header">etc</h2>
+        <div class="grid_container">
```
위에 있는 `.house_image_all`랑 요기 `.grid_container`도 사용되는 의미를 생각해보고 div 대신 시멘틱 태그로 바꿀 수 있으면 바꿔줘도 좋다고 생각합니당!!

Arooming commented on week1/index.html:
```diff
@@ -0,0 +1,76 @@
+      </section>
+    </main>
+
+    <footer class="footer"> 제주도 일기 끝 ㅋㅋ 
```
요기는 클래스를 주긴 했는데 css 파일 보니까 클래스를 사용하지 않고 `footer` 자체에 css 속성을 주고 있더라구 !
요거 말고도 그런 코드들이 몇몇 있는 것 같은데, 처음에 class를 다 지정해줬다가 css 속성을 정의할 때는 이걸 쓰지 않은 경우겠지?! 

리뷰 끝나고 코드 리팩토링때는 class를 지정해줬지만 생성자로 대체했거나 대체할 수 있는 태그들은 class 정의한거 지워줘도 좋을 것 같아!!

ExceptAnyone replied:
클래스를 지우는게 더 좋을지 vs css에 .footer로 수정하는게 더 나은지 모르겠어서 우선 css에 .footer로 수정했으!

Arooming commented on week1/index.html:
```diff
@@ -0,0 +1,76 @@
+
+    <footer class="footer"> 제주도 일기 끝 ㅋㅋ 
+      <br>앞으로 잘 부탁드립니다
+      <br>나는 css에 재능이 없다.
```
문단을 `<br>`로 구분하는건 좋지 않은 사례일 뿐만 아니라 스크린 리더를 사용해서 탐색하는 사용자에게도 문제가 된다고해! 
스크린 리더가 요소의 존재는 알려주지만 `<br>`에는 아무런 내용도 없기 때문에, 사용자가 내용이 어딨는지 몰라서 혼란을 느낄 수 있기 때문에..!!
앞으로는 `<p>`로 문단 구분 후, margin이나 padding 등등 `CSS 속성` 활용해서 간격 조절해주는 방향으로 해봅시당 !! 👊

Arooming commented on week1/style.css:
```diff
@@ -0,0 +1,162 @@
+.group-img {
+  width: 40rem;
+  height: 40rem;
+  margin: 5px;
```
웬만하면 css 단위는 통일해서 사용해주는게 좋다고 하더라구! 
리팩토링할 때 단위 신경써주면 좋을 것 같고, px과 rem 차이는 내 코드에 리뷰로 남겨놨으니까 한 번 참고해줘잉!!

Arooming commented on week1/style.css:
```diff
@@ -0,0 +1,162 @@
+  text-align: center;
+}
+
+.nav_list a {
+  display: flex;
+  font-size: 30px;
+  justify-content: space-around;
+  align-items: center;
```
이 속성들을 li 태그 상위에 있는 `ul` 태그에 주면 가로로 정렬이 될거야!
아래 첨부한 사진처럼 상위 태그인 ul의 레이아웃에 따라 li 태그 위치의 큰 틀이 잡힌다고 생각하면 이해하기 조금 쉬울 것 같은데, 혹시 이해가 됐을까아아! 🫨 
이해하기 어려운 부분이 있다면 리뷰 한 번 더 남겨주거나 편하게 얘기해조!!!

<img width="392" alt="스크린샷 2023-10-16 오후 4 28 12" src="https://github.com/DO-SOPT-WEB/JeonganJang/assets/80264647/0a26085f-d4dc-4f30-942c-2b05d59461e0">

Arooming replied:
ul 태그에 `list-style: none;` 속성을 적용해주면 하위 li 태그들에 앞에 붙어있는 목차 마크들이 사라져! 
그리고 목차 태그들은 왼쪽 앞에 자동으로 `padding-left` 값이 들어가 있으니까 맨 앞에 의문의 간격을 없애고 싶다면 `padding-left: 0`일케 해주면 된당!!

ExceptAnyone replied:
오 신기방기 ㅋㅋㅋㅋ 이해 완전 됐으!

Arooming commented on week1/style.css:
```diff
@@ -0,0 +1,162 @@
+}
+
+.food_image {
+  width: 30rem;
+  height: 30rem;
+  margin: 5px;
```
요기쪽도 그렇고 단위가 하나로 통일되지 않고 섞여있는 부분들이 꽤 있는 것 같은데 그런 부분 리팩토링 때 한번 단위 통일 + 이 단위를 왜 사용해야 하는지 생각해보고 변경해보면 좋을 듯 합니당!!

Arooming commented on week1/style.css:
```diff
@@ -0,0 +1,162 @@
+
+/* 두 번째 섹션 */
+.second_content_header {
+  position: sticky;
+  top: 0;
+  background: skyblue;
+  padding: 10px;
+  justify-content: center;
+  z-index: 100;
```
css 속성도 정의하는 순서가 있다는거 알고 계신가요..!!
아래 내가 첨부한 사진이 정답도 아니고, 나중에 협업하거나 클라 팀이 생기게 되면 팀 내에서 따로 컨벤션을 정하긴 할거야!
그치만 혼자 css 정의할 때도 규칙을 정해놓고 하면 훨씬 가독성있는 코드가 되더라구! 그래서 나는 아래 첨부한 사진을 참고해서 최대한 컨벤션 지키면서 하려고 하는 편이야! 

다른 곳에서 정의한 컨벤션 찾아봐도 좋고, 오빠가 나름의 규칙을 가지고 컨벤션 정해도 좋으니까 최대한 컨벤션 지켜서 코드 구현해보는거 어떨까!

<img width="939" alt="KakaoTalk_Photo_2023-10-16-16-51-29" src="https://github.com/DO-SOPT-WEB/JeonganJang/assets/80264647/24a5430c-fe36-4d32-acd4-0788b2e3fcf3">

- 위의 사진에 대한 설명: `같은 의미`를 가지는 속성끼리 `한 묶음`으로 만들어주고, `다른 의미`를 가진 속성끼리는 `한 줄 공백`으로 나눠주기

ExceptAnyone replied:
아 나는 그냥 생각나는대로 적었는데 이런게 있었구나 땡큐땡큐!

Arooming reviewed (commented):
첫 과제!!!!! 하느라 너무너무 수고했어🤩🤩🤩
나도 첫 과제할 때 진짜 힘들었던 기억이 있는데, 가끔 못하겠다고 의기소침 하기도 했지만(?)ㅋㅋ
암튼 너무 열심히 잘 해준 것 같아!!! 최고다 😱
앞으로도 열심히 해보쟝!!!!!! 넘 수고햇어 제주도 가고싶다 나도 크크

moondda commented on week1/style.css:
```diff
@@ -0,0 +1,162 @@
+
+.nav_list:hover {
+  color: skyblue;
+  transition: all 250ms ease-out;
```
오홍 transition이용해서 천천히 바뀌게 해줬네! 필수구현사항에는 없었는데 구냥 구현해본거야?? 부드럽게 변환되는 것도 예쁜거같넹

ExceptAnyone replied:
엉 그냥 해본거야 ㅋㅋㅋㅋ 저거라도 이쁘게 해보자 하면서 ㅋㅋㅋㅋㅋㅋ

ExceptAnyone committed:
refactor: 코드 리뷰 반영

section 태그 안에서 가장 바깥 쪽 div를 article로 변경.
이미지태그 alt속성 조금 더 자세히 설명
br태그 대신 p태그로 변경.
nav버튼 가로로 정렬

seobbang commented on week1/index.html:
```diff
@@ -0,0 +1,80 @@
+      <h1 class="header-title">장정안의 제주도 일기</h1>
+    </header>
+    <main>
+      <section class="group">
```
class를 줄 때는 좀 더 의미를 가지는 이름으로 해주면 좋을 것 같아요! 예를 들자면 그냥 group 보다는 어떤 그룹인지! **preview-group** 이런식으로 주어야 아무래도 선택자를 활용해서 CSS JS를 짤 때나, 가독성 측면에서도 훨씬 좋겠죠?!

seobbang commented on week1/index.html:
```diff
@@ -0,0 +1,80 @@
+    </header>
+    <main>
+      <section class="group">
+        <img src="./img/img5.JPG" alt="숙소 안 단체" class="group-img" />
```
alt 속성 야무지게 다 넣어줬네!! 최고 👍

seobbang commented on week1/index.html:
```diff
@@ -0,0 +1,80 @@
+      <p>앞으로 잘 부탁드립니다</p>
+      <p>나는 css에 재능이없다.</p>
+    </footer>
+    <button><a href="#">클릭하면 맨 위로!</a></button>
```
a만 있어도 그 자체로 링크 이동이 가능해요! 혹시 button으로 한번 더 감싼 이유가 있을까요?

seobbang reviewed (commented):
고생하셨습니당 ~! 나중에 시간되면 심화과제도 도전해보기!!!
